lib/block_view_lockups.go

```diff
@@ -739,16 +1075,36 @@ func (bav *UtxoView) _connectCoinLockup(
 			errors.Wrap(RuleErrorCoinLockupInvalidLockupDuration, "_connectCoinLockup")
 	}
 
-	// Compute the lockup duration in nanoseconds.
-	lockupDurationNanoSeconds := txMeta.UnlockTimestampNanoSecs - blockTimestampNanoSecs
+	// Validate the vesting end timestamp as expiring at a logically valid time.
+	if txMeta.VestingEndTimestampNanoSecs < txMeta.UnlockTimestampNanoSecs {
+		return 0, 0, nil,
+			errors.Wrap(RuleErrorCoinLockupInvalidVestingEndTimestamp, "_connectCoinLockup")
+	}
 
-	// Determine the hodler PKID to use.
-	transactorPKIDEntry := bav.GetPKIDForPublicKey(txn.PublicKey)
-	if transactorPKIDEntry == nil || transactorPKIDEntry.isDeleted {
+	// In the vested case, validate that the underlying profile is the transactor.
+	if txMeta.VestingEndTimestampNanoSecs > txMeta.UnlockTimestampNanoSecs &&
+		!reflect.DeepEqual(txn.PublicKey, txMeta.ProfilePublicKey.ToBytes()) {
 		return 0, 0, nil,
-			errors.Wrap(RuleErrorCoinLockupInvalidTransactorPKID, "_connectCoinLockup")
+			errors.Wrapf(RuleErrorCoinLockupInvalidVestedTransactor, "_connectCoinLockup: Profile "+
+				"pub key: %v, signer public key: %v", PkToString(txn.PublicKey, bav.Params),
+				PkToString(txn.PublicKey, bav.Params))
 	}
-	hodlerPKID := transactorPKIDEntry.PKID
+
```

Comment:
I want you to add a deep sanity-check as follows:
1. Near the top of this function, after you have everyone's pkids, snapshot the balances of the transactor, the recipient, and the creator.
2. Toward the bottom, pull the three balances again AND compute the delta from (sum(newLockedBlanaceEntries) - sum(prevLockedBalanceEntries))
3. Verify that sum(newBalances) + sum(lockedBalanceEntryDelta) <= sum(oldBalances)

Reply:
We do a similar check for NFTs and DAO Coin Limit Orders btw. Those are more hardcore than what I'm suggesting here but you can look at them.